core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -706,7 +706,7 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
       // Here, we'll copy the data into our data pages. Because we only store a relative offset from
       // the key address instead of storing the absolute address of the value, the key and value
       // must be stored in the same memory page.
-      // (8 byte key length) (key) (value) (8 byte pointer to next value)
+      // (uao: klen + vlen + uaolen) (uao: klen) (key) (value) (8 byte pointer to next value)
```

Comment:
what does `(uao: klen + vlen + uaolen) (uao: klen)` mean?

Reply:
it's `(total length) (key length) ...`?

Reply:
yes, if `total length` here doesn't include the first `uaoSize`.

Reply:
How about using the similar description at lines 56-61?

Reply:
It will be a little bit redundant? Maybe just, `(total length) (key length) (key) (value) (8 byte pointer to next value)` ?

Reply:
SGTM